core/src/main/java/com/oath/oak/Handle.java

```diff
@@ -39,16 +39,24 @@ boolean isDeleted() {
         return deleted.get();
     }
 
-    boolean remove(MemoryManager memoryManager) {
+    <V> Result<V> remove(MemoryManager memoryManager, V oldValue, Function<ByteBuffer, V> transformer) {
+        Result<V> res = Result.withFlag(false);
         writeLock.lock();
-        if (isDeleted()) {
+        try {
+            if (!isDeleted()) {
+                V vv = (transformer != null) ? transformer.apply(value) : null;
```

Comment:
Not a necessary, but still... Can there be an "empty transformer" that does nothing and returns null for ZC case? This way we may avoid countless checks for transformer being null...

Reply:
That's a good idea, although the checks are there because we return different *types* (`Result.withValue` vs `Result.withFlag`). A different abstraction like subclassing deeper for ZC/nonZC cases could fit.  I refrained from further abstraction to limit the size of this change and possible impact on performance.